src/components/sidebar/routePathView/RoutePathCopySegmentView.tsx

```diff
@@ -0,0 +1,198 @@
+            .find(link => link.endNode.id === copyStartNodeId)!.orderNumber + 1;
+            for (let i = 0; i < segmentsToCopy.length; i += 1) {
+                await this.copySegment(segmentsToCopy[i].routePathLinkId, orderNumber);
+                orderNumber += 1;
+            }
+        } else if (isOppositeDirection) {
+            const orderNumber = this.props.routePathStore!.routePath!.routePathLinks!
+                .find(link => link.startNode.id === copyEndNodeId)!.orderNumber;
+            for (let i = segmentsToCopy.length - 1; i >= 0; i -= 1) {
+                await this.copySegment(segmentsToCopy[i].routePathLinkId, orderNumber);
+            }
+        } else {
+            // sanity check
+            throw 'Node with odd neighbors not found from current routePath by startNodeId or endNodeId.'; // tslint:disable-line max-line-length
+        }
+
+        this.props.routePathCopySegmentStore!.clear();
+        this.props.toolbarStore!.selectTool(null);
+
+        this.props.dialogStore!.closeLoaderMessage();
+        this.props.dialogStore!.setFadeMessage('Segmentti kopioitu!', DialogType.Success);
+    }
+
+    private copySegment = async (routePathLinkId: number, fixedOrderNumber: number) => {
+        const routePathLink: IRoutePathLink = await RoutePathLinkService
+            .fetchRoutePathLink(routePathLinkId);
+        routePathLink.orderNumber = fixedOrderNumber;
+
+        this.props.routePathStore!.addLink(routePathLink);
+    }
+
+    private openRoutePathInNewTab = (routePath: ICopySegmentRoutePath) => () => {
+        const routePathLink = routeBuilder
+        .to(SubSites.routePath)
+        .toTarget([
+                routePath.routeId,
+                Moment(routePath.startTime).format('YYYY-MM-DDTHH:mm:ss'),
+                routePath.direction,
+        ].join(','))
+        .toLink();
+        window.open(routePathLink, '_blank');
+    }
+
+    private renderErrorMessage = () => {
+        return (
+            <div className={s.routePathList}>
+                <div className={s.messageContainer}>
+                    Virhe alku- ja loppusolmun asetuksessa.
+                </div>
+            </div>
+        );
+    }
+
+    private renderResults = () => {
+        const routePaths = this.props.routePathCopySegmentStore!.routePaths;
+        return (
+            <div className={s.routePathList}>
+                { routePaths.length === 0 ? (
+                    <div className={s.messageContainer}>
+                        Kopioitavia reitinsuunnan segmenttejä ei löytynyt valitulta väliltä.
+                        Kokeile asettaa pienempi kopioitava väli.
+                    </div>
+                ) : (
+                    <>
+                    { routePaths.map((routePath, index) => {
+                        return this.renderRoutePathRow(
+                            routePath, `row-${index}-${routePath.routeId}'`);
```

Comment:
The string ends with a '
Suppose it is a mistake?

Reply:
Mistake yes, but I guess it didn't cause any bugs. Fixed it tho.